pl_bolts/datamodules/base_datamodule.py

```diff
@@ -0,0 +1,152 @@
+import os
+from abc import abstractmethod
+from typing import Optional, Tuple, Union
+
+import torch
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, random_split
+
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+try:
+    from torchvision import transforms as transform_lib
+except ModuleNotFoundError:
+    warn_missing_pkg("torchvision")  # pragma: no-cover
+    _TORCHVISION_AVAILABLE = False
+else:
+    _TORCHVISION_AVAILABLE = True
+
+
+class BaseDataModule(LightningDataModule):
```

Comment:
if it is refactor for Vision lets call it so and do not confuse users with another base level
```suggestion
class VisionDataModule(LightningDataModule):
```

Reply:
btw, is it any vision or `torchvision` only?

Reply:
It is for the basic torchvision datasets with the same API (CIFAR10, FashionMNIST, MNIST)